.github/workflows/build-publish.yml

```diff
@@ -0,0 +1,199 @@
+          baseimage: 'mcr.microsoft.com/windows/nanoserver:ltsc2019'
+          tagext: 'nanoserver2019'
+          canpatch: false
+        - os: windows-2022
+          file: ./builder/Dockerfile.windows
+          baseimage: 'mcr.microsoft.com/windows/nanoserver:ltsc2022'
+          tagext: 'nanoserver2022'
+          canpatch: false
+    defaults:
+      run:
+        shell: pwsh
+    steps:
+    - uses: actions/checkout@v3
+    # Perform a 'docker login' so that we can push to the ACR that provides the MCR images.
+    # This requires an Az login first.
+    - name: Az CLI Login
+      uses: azure/login@92a5484dfaf04ca78a94597f4f19fea633851fa2 # v1.4.6
+      with:
+        client-id: ${{ vars.AZURE_CLIENT_ID }}
+        tenant-id: ${{ vars.AZURE_TENANT_ID }}
+        subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+    - name: Authenticate to ACR
+      run: az acr login -n ${{ vars.AZURE_REGISTRY_SERVER }}
+    # Perform a 'docker login' so that we can push to the current repo's GHCR. We will push the unpatched
+    # images to GHCR first, run a Trivy scan and Copa patch on those, and then push the patched images
+    # to both GHCR and ACR.
+    - name: Authenticate to GHCR
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    # Work out the target image references (registry images and tags) for publishing.
+    - name: Get Image References
+      id: getref
+      run: |
+        $tag_id = "${{ needs.common.outputs.tag_id_base }}-${{ matrix.tagext }}"
+        echo "tag_id=$tag_id" >> $env:GITHUB_OUTPUT
+        echo "acr_image_ref=${{ needs.common.outputs.acr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "ghcr_image_ref=${{ needs.common.outputs.ghcr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "${{ matrix.os }}-acr-image-ref=${{ needs.common.outputs.acr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "${{ matrix.os }}-ghcr-image-ref=${{ needs.common.outputs.ghcr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+    - name: Build Image
+      id: build
+      run: docker build -f ${{ matrix.file }} --build-arg BASE_IMAGE=${{ matrix.baseimage }} -t ${{ steps.getref.outputs.ghcr_image_ref }} .
+    # If this OS supports patching images using Copa, push the unpatched image to GHCR with the tag suffix '-unpatched'.
+    # Then, scan and patch this image, and re-tag it with the original tag (no suffix).
+    - name: Push Unpatched Image to GHCR
+      if: matrix.canpatch
+      run: |
+        docker tag ${{ steps.getref.outputs.ghcr_image_ref }} ${{ steps.getref.outputs.ghcr_image_ref }}-unpatched
+        docker push ${{ steps.getref.outputs.ghcr_image_ref }}-unpatched
+    - name: Generate Trivy Report
+      if: matrix.canpatch
+      uses: aquasecurity/trivy-action@41f05d9ecffa2ed3f1580af306000f734b733e54 # v0.11.2
```

Comment:
make sure to add dependabot if you are pinning otherwise these will be stale over time

Reply:
Just trying to work out what we'd need to enable for this to work. It seems dependabot is paused on this repo because we haven't merged its PRs for too long. So I guess we'd need to do that. But also maybe we need this?
<img width="377" alt="image" src="https://github.com/Azure/aks-periscope/assets/1817696/e7d69134-b0d7-49a5-a988-e5473d829937">
cc: @Tatsinnit - in case you know more about this than me...

Reply:
💡 Oh, yeah. How about we create tracking issue for this repo, separately for enabling this for dependent bot, given all the good work under this PR can go ahead and we can enable that dependent bot alert post this PR merge? How does that feel to all. thank you so much for this. ❤️ cc: @sozercan + @peterbom

![](https://media.giphy.com/media/MZQkUm97KTI1gI8sUj/giphy.gif)

Reply:
Created #234 for this.